src/utils.cpp

```diff
@@ -107,25 +108,21 @@ int get_word_end(const char* text, int start) {
 }
 
 std::optional<std::string> read_file_to_string(const char* path) {
-    FILE* f = fopen(path, "r");
-    if (!f) return std::nullopt;
+    std::ifstream input_stream {path, std::fstream::in | std::fstream::ate};
+    if (!input_stream) return std::nullopt;
 
-    fseek(f, 0, SEEK_END);
-    size_t size = ftell(f);
+    const std::size_t size = input_stream.tellg();
```

Comment:
I think it would be more modern and safer to use `fs::file_size()` here.

Reply:
I'll fix this later today!